### PR TITLE
Krzeelzb assignment 3

### PR DESCRIPTION
**Pull request by krzeelzb**
ready assignment 3

### CODE REVIEW

File: krzeelzb/assignment-3/src/Grid.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+    public static int getColumnSize(){
+        return grid.get(0).size();
+    }
+
+    public ArrayList<ArrayList<String>> getGrid() {
+        return grid;
+    }
+
+    public void setGrid(ArrayList<ArrayList<String>> grid) {
```
Comment (imorlowska) [inline krzeelzb/assignment-3/src/Grid.java]: you could set this is the Grid constructor.

File: krzeelzb/assignment-3/src/Lexicon.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import java.util.List;
+import java.util.stream.Stream;
+
+public class Lexicon {
+    List<String> words;
```
Comment (imorlowska) [inline krzeelzb/assignment-3/src/Lexicon.java]: The order of the words doesn't matter - using a HashSet<String> would be more efficient.

File: krzeelzb/assignment-3/src/Lexicon.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    public Boolean isWord(String word){
+        return words.contains(word);
+    }
+
+    public Boolean isPrefix(String word){
+       return words.stream()
```
Comment (imorlowska) [inline krzeelzb/assignment-3/src/Lexicon.java]: Pre-computing a hash set of prefixes when we create the grid object would result in faster isPrefix lookup.

File: krzeelzb/assignment-3/src/Lexicon.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+               .anyMatch(s->s.startsWith(word));
+    }
+
+    public Boolean isSufix(String word){
```
Comment (imorlowska) [inline krzeelzb/assignment-3/src/Lexicon.java]: ?

File: krzeelzb/assignment-3/src/WordFinder.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        put("down-left", new ArrayList<Integer>(Arrays.asList(1, -1)));
+        put("down", new ArrayList<Integer>(Arrays.asList(1, 0)));
+        put("down-right", new ArrayList<Integer>(Arrays.asList(1, 1)));
+    }};
+
+    private void findWords(Grid grid, Lexicon lexicon, int row, int column, String word) {
+        String stringToCheck = word + grid.getGrid().get(row).get(column).toUpperCase();
+
+        if (lexicon.isWord(stringToCheck)) {
+            if (!foundWords.contains(stringToCheck))
```
Comment (imorlowska) [inline krzeelzb/assignment-3/src/WordFinder.java]: if you use a hash set you don't need to worry about checking if the word was already found

File: krzeelzb/assignment-3/src/WordFinder.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+            int neighbourRow = row + values.get(0);
+            int neighbourColumn = column + values.get(1);
+
+            if ((row + values.get(0)) >= 0 && (column + values.get(1)) >= 0) {
+                if ((row + values.get(0)) < rowsNumber) {
+                    if ((column + values.get(1)) < columnsNumber) {
+                        if (grid.getGrid().get(row + values.get(0)).get(column + values.get(1)) != null) {
+                            if (lexicon.isPrefix(stringToCheck)) {
+                                prefixesList.add(stringToCheck);
+                                findWords(grid, lexicon, neighbourRow, neighbourColumn, stringToCheck);
```
Comment (imorlowska) [inline krzeelzb/assignment-3/src/WordFinder.java]: Do you keep track which fields were already visited? For example if you had a grid { O, H } and lexicon: O, OH, HO, OHO, OHOHOHOHO only words O and OH and HO should be returned.

File: krzeelzb/assignment-3/src/test/TestClass.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class TestClass {
```
Comment (imorlowska) [inline krzeelzb/assignment-3/src/test/TestClass.java]: In future assignments please add more extensive tests. Try thinking of corner cases!
